akka-persistence/src/test/scala/akka/persistence/PersistentActorBoundedStashingSpec.scala

```diff
@@ -90,11 +101,23 @@ class ThrowExceptionStrategyPersistentActorBoundedStashingSpec
 
       //internal stash overflow
       1 to (2 * capacity) foreach (persistentActor ! Cmd(_))
-      //after PA stopped, all stashed messages forward to deadletters
-      1 to capacity foreach (i ⇒ expectMsg(DeadLetter(Cmd(i), testActor, persistentActor)))
-      //non-stashed messages
-      (capacity + 2) to (2 * capacity) foreach (i ⇒ expectMsg(DeadLetter(Cmd(i), testActor, persistentActor)))
 
+      //after PA stopped, all stashed messages forward to deadletters
+      //the message triggering the overflow is lost, so we get one less message than we sent
+      val messages: Seq[DeadLetter] = 1 until (2 * capacity) map { _ ⇒ expectMsgType[DeadLetter] }
+
+      //if the mailbox of the PA happens to close before we manage to send all messages, then those
+      //messages can get sent to dead letters before or mixed with the stashed ones so we need to partition them
+      val (stashed, nonStashed) = messages.partition {
+        case DeadLetter(Cmd(i: Int), `testActor`, `persistentActor`) ⇒ i <= capacity
+      }
+
+      checkMessagesAreInOrder(stashed, 1)
+      //since we don't know exactly which message is the last one going into the mailbox and which one is the
+      //first one going directly to dead letters, we can't assume an ordering and need to sort the messages
+      checkMessagesAreInOrder(nonStashed.sortBy {
+        case DeadLetter(Cmd(i: Int), _, _) ⇒ i
+      }, capacity + 2)
```

Comment:
Is it possible to simplify the test, I'm not sure what it does.
We know exactly how many messages that are required for the stash overflow to happen? can we just verify that the actor is stopped after that? this checking of deadletters is not really a concern of this test, is it?